pkg/sql/catalog/bootstrap/metadata.go

```diff
@@ -187,152 +194,89 @@ func (ms MetadataSchema) GetInitialValues() ([]roachpb.KeyValue, []roachpb.RKey)
 	return ret, splits
 }
 
+// InitialUserDescriptorID returns the smallest descriptor ID for a non-system
+// descriptor. This value is used to initialize the descriptor ID sequence.
+func (ms MetadataSchema) InitialUserDescriptorID() descpb.ID {
+	return descpb.ID(keys.MaxReservedDescID) + 1
+}
+
 // DescriptorIDs returns the descriptor IDs present in the metadata schema in
 // sorted order.
 func (ms MetadataSchema) DescriptorIDs() descpb.IDs {
 	descriptorIDs := descpb.IDs{}
-	for _, md := range ms.descs {
-		descriptorIDs = append(descriptorIDs, md.desc.GetID())
+	for _, d := range ms.descs {
+		descriptorIDs = append(descriptorIDs, d.GetID())
 	}
 	sort.Sort(descriptorIDs)
 	return descriptorIDs
 }
 
-// systemTableIDCache is used to accelerate name lookups on table descriptors.
-// It relies on the fact that table IDs under MaxReservedDescID are fixed.
-//
-// Mapping: [systemTenant][tableName] => tableID
-var systemTableIDCache = func() [2]map[string]descpb.ID {
-	cacheForTenant := func(systemTenant bool) map[string]descpb.ID {
-		cache := make(map[string]descpb.ID)
-
-		codec := keys.SystemSQLCodec
-		if !systemTenant {
-			codec = keys.MakeSQLCodec(roachpb.MinTenantID)
-		}
-
-		ms := MetadataSchema{codec: codec}
-		addSystemDescriptorsToSchema(&ms)
-		for _, d := range ms.descs {
-			t, ok := d.desc.(catalog.TableDescriptor)
-			if !ok || t.GetParentID() != keys.SystemDatabaseID || t.GetID() > keys.MaxReservedDescID {
-				// We only cache table descriptors under 'system' with a
-				// reserved table ID.
-				continue
-			}
-			cache[t.GetName()] = t.GetID()
-		}
-
-		return cache
-	}
-
-	var cache [2]map[string]descpb.ID
-	for _, b := range []bool{false, true} {
-		cache[boolToInt(b)] = cacheForTenant(b)
-	}
-	return cache
-}()
-
-func boolToInt(b bool) int {
-	if b {
-		return 1
-	}
-	return 0
-}
-
-// LookupSystemTableDescriptorID uses the lookup cache above
-// to bypass a KV lookup when resolving the name of system tables.
-func LookupSystemTableDescriptorID(
-	codec keys.SQLCodec, dbID descpb.ID, tableName string,
-) descpb.ID {
-	if dbID != systemschema.SystemDB.GetID() {
-		return descpb.InvalidID
-	}
-
-	systemTenant := boolToInt(codec.ForSystemTenant())
-	dbID, ok := systemTableIDCache[systemTenant][tableName]
-	if !ok {
-		return descpb.InvalidID
-	}
-	return dbID
-}
-
```

Comment:
`LookupSystemTableDescriptorID` has been replaced by stuff in the `system_descriptors.go` file.